### ISSUE TITLE
Add WebSocket Support to DiceDB

### ISSUE THREAD

**Issue by pratikpandey21** [DiceDB/dice#555]
### Description

Extend the existing HTTP server logic to add WebSocket server support. This enhancement will enable real-time, bidirectional communication between clients and the server, allowing for more interactive and dynamic use cases, such as live data updates and push notifications.

Refer to - https://github.com/DiceDB/dice/pull/512 and https://github.com/DiceDB/dice/pull/527.

**Comment by jainam2907** [comment]
I would like to work on this issue.

**Comment by lucifercr07** [comment]
@jainam2907 this issue is dependent with #556, please collabarate with @Saumay on this.
Thanks for contributing, assigned.

**Comment by Saumay** [comment]
Hey @jainam2907 , how is it going for the WS support?
I am working on #556, which is dependent on this.
Would you want to connect sometime regarding this?

**Comment by lucifercr07** [comment]
@jainam2907 any updates from your side, please let us know if any blockers.

**Comment by psrvere** [comment]
@lucifercr07 @Saumay - I will happy to pick this up if it's available.

**Comment by lucifercr07** [comment]
> @lucifercr07 @Saumay - I will happy to pick this up if it's available.

@psrvere Let's wait for @jainam2907 to respond till today, if not will reassign.

**Comment by jainam2907** [comment]
Apologies for the delay. Due to some personal commitments, I haven't been able to work on this yet. I can start in a couple of days. If other issues are getting blocked by this, please feel free to reassign it to someone else.

**Comment by lucifercr07** [comment]
@jainam2907 no issues, since other dependent issues are blocked on this will reassign it for now. @psrvere would you like to pick this up?

**Comment by psrvere** [comment]
Yes @lucifercr07 - Please assign to me.

**Comment by pratikpandey21** [comment]
Thanks @psrvere . Please let us know in case you need any help!

**Comment by gauravrp** [comment]
@psrvere Are you actively working on this?
Would love to collaborate on it

**Comment by psrvere** [comment]
Hey @gauravrp, thanks for showing interest. 
Yes I am almost done with writing the code, only writing tests and testing is left. 
Try to pick up another issue preferably with `dificulty-level -- easy` since you are new to the project.

**Comment by psrvere** [comment]
@Saumay - FYI. PR has been raised and under review.

**Comment by Saumay** [comment]
Thanks @psrvere!
